src/core/evaluator.js

```diff
@@ -3465,36 +3457,30 @@ class PartialEvaluator {
       isType3Font: false,
     };
 
-    var cMapPromise;
     if (composite) {
-      var cidEncoding = baseDict.get("Encoding");
+      const cidEncoding = baseDict.get("Encoding");
       if (isName(cidEncoding)) {
         properties.cidEncoding = cidEncoding.name;
       }
-      cMapPromise = CMapFactory.create({
+      const cMap = await CMapFactory.create({
         encoding: cidEncoding,
         fetchBuiltInCMap: this._fetchBuiltInCMapBound,
         useCMap: null,
-      }).then(function (cMap) {
-        properties.cMap = cMap;
-        properties.vertical = properties.cMap.vertical;
       });
-    } else {
-      cMapPromise = Promise.resolve(undefined);
+      properties.cMap = cMap;
+      properties.vertical = properties.cMap.vertical;
     }
 
-    return cMapPromise
-      .then(() => {
-        return this.extractDataStructures(dict, baseDict, properties);
-      })
-      .then(newProperties => {
+    return this.extractDataStructures(dict, baseDict, properties).then(
+      newProperties => {
         this.extractWidths(dict, descriptor, newProperties);
 
         if (type === "Type3") {
           newProperties.isType3Font = true;
         }
         return new Font(fontName.name, fontFile, newProperties);
-      });
+      }
+    );
```

Comment:
Should this be better like this ?
```javascript
    const newProperties = await this.extractDataStructures(dict, baseDict, properties);
    this.extractWidths(dict, descriptor, newProperties);

    if (type === "Type3") {
      newProperties.isType3Font = true;
    }
    return new Font(fontName.name, fontFile, newProperties);
```

Reply:
Well, I don't think that what you're suggesting is necessarily *better*, but rather only *different*.
Given that it'd only serve to increase the total size of the diff, since there's more than one such "block", I'm not convinced that it's a necessary change.

Reply:
You're right, it's not necessary but seems to me more readable in only one block.
It was just a suggestion, you can skip my comment ;)